src/test/function_test/security/test_kms_client.cpp

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <absl/strings/str_split.h>
+#include <string.h>
+#include <memory>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "replica/kms_key_provider.h"
+#include "replica/replication_app_base.h"
+#include "test_util/test_util.h"
+#include "utils/error_code.h"
+#include "utils/errors.h"
+#include "utils/flags.h"
+
+namespace dsn {
+DSN_DECLARE_string(cluster_name);
+namespace security {
+DSN_DECLARE_bool(enable_acl);
+} // namespace security
+namespace replication {
+DSN_DECLARE_string(hadoop_kms_url);
+} // namespace replication
+} // namespace dsn
+
+class KmsClientTest : public pegasus::encrypt_data_test_base
```

Comment:
The test only check the functionality of when FLAGS_encrypt_data_at_rest is true IMO, the KMS client will not be used if it's disabled, right?

Reply:
> The test only check the functionality of when FLAGS_encrypt_data_at_rest is true IMO, the KMS client will not be used if it's disabled, right?

yes,the test will skip when FLAGS_hadoop_kms_url is empty  or FLAGS_encrypt_data_at_rest is false.

Reply:
It's not needed to derive from encrypt_data_test_base because it only test the case of  FLAGS_encrypt_data_at_rest = true.